java/vortex-spark/src/main/java/dev/vortex/spark/ArrowUtils.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.spark;
+
+import dev.vortex.relocated.org.apache.arrow.vector.types.DateUnit;
+import dev.vortex.relocated.org.apache.arrow.vector.types.FloatingPointPrecision;
+import dev.vortex.relocated.org.apache.arrow.vector.types.TimeUnit;
+import dev.vortex.relocated.org.apache.arrow.vector.types.pojo.ArrowType;
+import dev.vortex.relocated.org.apache.arrow.vector.types.pojo.Field;
+import java.util.stream.Collectors;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+
+public final class ArrowUtils {
+    private ArrowUtils() {}
+
+    public static DataType fromArrowField(Field field) {
+        switch (field.getType().getTypeID()) {
+            case Struct:
+                return DataTypes.createStructType(field.getChildren().stream()
+                        .map(child -> {
+                            DataType dt = fromArrowField(child);
+                            return new StructField(child.getName(), dt, child.isNullable(), Metadata.empty());
```

Comment:
I see that the Scala code doesn't do this but it seems a little rude to not propagate the field's metadata. That said, it looks like Java wants a `Metadata` which we can only create by calling `child.getMetadata()` and copying that key-by-key into a `MetadataBuilder`.

Grr. Are we worried about the overhead of copying metadata for this method? I dunno. It just feels like the kind of thing on which a user would stub their toe.